src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2716,6 +2716,11 @@ public override BoundNode VisitDefaultExpression(BoundDefaultExpression node)
             return null;
         }
 
+        public override BoundNode VisitUnboundObjectCreationExpression(UnboundObjectCreationExpression node)
+        {
+            return null;
```

Comment:
>return null; [](start = 12, length = 12)

should probably throw instead

Reply:
Can happen in error cases. (same as a few above)

Reply:
Could you expound a bit? 
I would expect that every expression gets forced to a target-type or natural type, which means the bound node would become an `BoundObjectCreationExpression` in all cases (although sometimes with an error type).

Reply:
we'll need to add BindToNaturalType in more places but it's very subtle because we need an extensive coverage to catch all the cases and still not a guarantee that this won't happen with some combination of nodes.

Reply:
Hum, can we add an assertion instead, then? If we did reach here, it would mean that we have a bug (lacking a call to `BindToNaturalType` somewhere)